src/client/options/resolver_config.rs

```diff
@@ -0,0 +1,43 @@
+use trust_dns_resolver::config::ResolverConfig as TrustDnsResolverConfig;
+
+/// Configuration for the upstream nameservers to use for resolution.
+///
+/// This is a thin wrapper around a `trust_dns_resolver::config::ResolverConfig` provided to ensure
+/// API stability.
+#[derive(Clone, Debug, PartialEq)]
+pub struct ResolverConfig {
+    pub(crate) inner: TrustDnsResolverConfig,
+}
+
+impl ResolverConfig {
```

Comment:
I opted to provide just the predefined configs rather than exposing further functionality since that is likely the only things our users need and it could get hard to maintain the additional API surface. (see [`trust-dns-resolver` docs](https://docs.rs/trust-dns-resolver/0.20.2/trust_dns_resolver/config/struct.ResolverConfig.html) for the full type).